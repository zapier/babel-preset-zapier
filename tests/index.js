/* eslint-env jest */
const babel = require('babel-core');
const omit = require('lodash.omit');
const buildPreset = require('../index');

const transform = (code, options, fakeEnvironment) => {
  const presetConfig = buildPreset(null, options);

  const babelConfig = Object.assign(
    presetConfig,
    {
      babelrc: false, // don't use .babelrc
      env: omit(presetConfig.env, 'test'), // get rid of test env, otherwise we can't test non-test env scenarios
    },
    presetConfig.env[fakeEnvironment] || {}
  );

  return babel.transform(code, babelConfig).code;
};

describe('babel-preset-zapier', () => {
  it("doesn't compile ES modules to commonjs by default", () => {
    const code = `
      import Foo from 'foo';
    `;

    expect(transform(code)).toMatchSnapshot();
  });

  it('compiles to commonjs modules if `modules: true` option is passed', () => {
    const code = `
      import Foo from 'foo';
    `;

    expect(transform(code, { modules: true })).toMatchSnapshot();
  });

  it('strips flow annotations', () => {
    const code = `
        type Foo = {
          bar: string
        };

        function greet(name: string) {}
      `;

    expect(transform(code)).toMatchSnapshot();
  });

  it('compiles JSX', () => {
    const code = `
      <Foo />;
    `;

    expect(transform(code)).toMatchSnapshot();
  });

  // This test will break when we drop IE11 support
  it('compiles `const` to `var`', () => {
    const code = `
      const foo = 'bar';
    `;

    expect(transform(code)).toMatchSnapshot();
  });

  it('doesnt strip proptypes when not in prod env', () => {
    const code = `
      const Baz = (props) => (
        <div {...props} />
      );

      Baz.propTypes = {
        className: PropTypes.string
      };
    `;

    expect(transform(code)).toMatchSnapshot();
  });

  it('strips proptypes when in prod env', () => {
    const code = `
      const Baz = (props) => (
        <div {...props} />
      );

      Baz.propTypes = {
        className: PropTypes.string
      };
    `;

    expect(transform(code, {}, 'production')).toMatchSnapshot();
  });
});
