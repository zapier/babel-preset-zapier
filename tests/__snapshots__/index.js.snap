// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-preset-zapier compiles JSX 1`] = `
"\\"use strict\\";

React.createElement(Foo, null);"
`;

exports[`babel-preset-zapier compiles to commonjs modules 1`] = `
"\\"use strict\\";

var _foo = _interopRequireDefault(require(\\"foo\\"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }"
`;

exports[`babel-preset-zapier doesn't compile ES modules to commonjs by default 1`] = `
"
import Foo from 'foo';"
`;

exports[`babel-preset-zapier doesnt strip proptypes when not in prod env 1`] = `
"\\"use strict\\";

const Baz = props => React.createElement(\\"div\\", props);

Baz.propTypes = {
  className: PropTypes.string
};"
`;

exports[`babel-preset-zapier strips flow annotations 1`] = `
"\\"use strict\\";

function greet(name) {}"
`;

exports[`babel-preset-zapier when on production env strips proptypes when in prod env 1`] = `
"\\"use strict\\";

const Baz = props => React.createElement(\\"div\\", props);"
`;
